com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -391,6 +391,8 @@ internal void SendSpawnCallForObject(ulong clientId, NetworkObject networkObject
                     var size = bufferSizeCapture.StopMeasureSegment();
                     NetworkManager.NetworkMetrics.TrackObjectSpawnSent(clientId, networkObject.NetworkObjectId, networkObject.name, size);
                 }
+
+                networkObject.MarkVariablesDirty();
```

Comment:
After spawning an object, we mark all its variable dirty. Trying to separate peas and carrots. Spawning is a thing. The value update is another.

Reply:
wasn't inital netvars written to the payload next to spawn message?

Reply:
Yes, sharp eyes. :-) That will be the final piece:

```
--- a/com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs
+++ b/com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs
@@ -453,8 +453,6 @@ namespace Unity.Netcode
                 var (isReparented, latestParent) = networkObject.GetNetworkParenting();
                 NetworkObject.WriteNetworkParenting(writer, isReparented, latestParent);
             }
-
-            networkObject.WriteNetworkVariableData(writer.GetStream(), clientId);
         }
```

It is not efficient to have such a large spawn message. The spawn itself is just "spawn this". That way, it can get efficiently stored, sent, ack'ed, re-sent, etc... by the snapshot system.
A different message, after the spawn, will carry the value update. It seems in current status, this separation works fine excpet for those Network{List,Set,Dictionary}. So, this PR is not ready. Still in progress.

Reply:
To clarify, this change-to-be-done-later above, makes it symmetric. With it, both snapshot and non-snapshot have the spawns and the value updates separated.

Reply:
so, are you saying that we're going to get rid of that initial netvar write block that writes into one single payload and have only the spawn message first, mark things dirty and leave the rest of the things up to netvar replication (whether snapshotted or not) to eventually update props on the clients later? if so, I do support that direction :) however, I wonder how many thing will break simply because of expecting initial netvar values to be there at the same time when we spawn the networkobject — let's break them by doing the right thing and fix them later when they come up!